src/FlexibleHttpClient.php

```diff
@@ -11,7 +11,7 @@
  *
  * @author Joel Wurtz <joel.wurtz@gmail.com>
  */
-class FlexibleHttpClient implements HttpClient, HttpAsyncClient
+final class FlexibleHttpClient implements HttpClient, HttpAsyncClient
```

Comment:
Don't know, rename is discussed in #16, although no better name came up yet IMO.